waypoint/services/nats_service.py

```diff
@@ -0,0 +1,148 @@
+                }
+            else:
+                logger.trace("Tenant call got no group_id")
+                subscribe_kwargs = {
+                    "subject": f"{NATS_SUBJECT}.*.{wallet_id}",
+                    "stream": NATS_STREAM,
+                }
+            subscription = await self.js_context.pull_subscribe(**subscribe_kwargs)
+
+            return subscription
+
+        except BadSubscriptionError as e:
+            logger.error(f"BadSubscriptionError subscribing to NATS: {e}")
+            raise
+        except Error as e:
+            logger.error(f"Error subscribing to NATS: {e}")
+            raise
+        except Exception:
+            logger.exception("Unknown error subscribing to NATS")
+            raise
+
+    @asynccontextmanager
+    async def process_events(
+        self,
+        group_id: str,
+        wallet_id: str,
+        topic: str,
+        stop_event: asyncio.Event,
+        duration: int = 10,
+    ):
+        logger.debug(
+            f"Processing events for group {group_id} and wallet {wallet_id} on topic {topic}"
+        )
+
+        subscription = await self._subscribe(group_id=group_id, wallet_id=wallet_id)
+
+        async def event_generator():
+            end_time = time.time() + duration
+            while not stop_event.is_set():
+                remaining_time = remaining_time = end_time - time.time()
+                logger.trace(f"remaining_time: {remaining_time}")
+                if remaining_time <= 0:
+                    logger.debug("Timeout reached")
+                    stop_event.set()
+                    break
+
+                try:
+                    messages = await subscription.fetch(10, 1)
+                    for message in messages:
+                        if message.headers.get("event_topic") == topic:
+                            event = json.loads(message.data)
+                            yield CloudApiWebhookEventGeneric(**event)
+                        await message.ack()
+                except TimeoutError:
+                    logger.trace("Timeout fetching messages continuing...")
+                    await asyncio.sleep(1)
+
+        try:
+            yield event_generator()
+        except asyncio.CancelledError:
+            logger.debug("Event generator cancelled")
+            stop_event.set()
+        finally:
+            logger.trace("Closing subscription...")
+            await subscription.unsubscribe()
+            logger.debug("Subscription closed")
+
+    async def check_jetstream(self):
+        try:
+            account_info = await self.js_context.account_info()
+            is_working = account_info.streams > 0
+            logger.trace(f"JetStream check completed. Is working: {is_working}")
+            return {
+                "is_working": is_working,
+                "streams_count": account_info.streams,
+                "consumers_count": account_info.consumers,
+            }
+        except Exception:
```

Comment:
Just add `  # pylint: disable=W0718` to the `except Exception` line. To suppress pylint warnings

Please find all `except Exception` blocks in this PR and add pylint disable